src/rev_ai/customvocabulariesclient.py

```diff
@@ -0,0 +1,125 @@
+
+import requests
+import json
+from requests.exceptions import HTTPError
+# from .models import Job, Account, Transcript, CaptionType
+from . import __version__
+
+try:
+    from urllib.parse import urljoin
+except:
+    from urlparse import urljoin
+
+
+class RevAiCustomVocabulariesClient:
+    """Client which implements Rev.ai CustomVocabulary API"""
+
+    # Default version of Rev.ai
+    version = 'v1'
+
+    # Default address of the API
+    base_url = 'https://api.rev.ai/speechtotext/{}/vocabularies'.format(
+        version)
+
+    def __init__(self, access_token):
+        """Constructor
+
+        :param access_token: access token which authorizes all requests and
+                             links them to your account. Generated on the
+                             settings page of your account dashboard
+                             on Rev.ai
+        """
+        if not access_token:
+            raise ValueError('access_token must be provided')
+
+        self.default_headers = {
+            'Authorization': 'Bearer {}'.format(access_token),
+            'User-Agent': 'RevAi-PythonSDK/{}'.format(__version__)
+        }
+
+    def submit_custom_vocabularies(
+            self,
+            custom_vocabularies,
+            callback_url=None,
+            metadata=None):
+        """Submit custom vocabularies.
+
+        :param custom_vocabularies: List of objects containing phrases list
+        :param callback_url: callback url to invoke on job completion as a
+                             webhook
+        :param metadata: info to associate with the transcription job
+        """
+
+        if not custom_vocabularies:
+            raise ValueError('custom_vocabularies must be provided')
+
+        payload = self._create_custom_vocabularies_options_payload(
+            callback_url,
+            metadata,
+            custom_vocabularies
+        )
+
+        response = self._make_http_request(
+            "POST",
+            self.base_url,
+            json=payload
+        )
+
+        return response.json()
+
+    def get_custom_vocabularies(
+            self,
+            id):
+        """ Get the custom vocabulary status
+        :param id: string id of custom vocabulary submission
+        """
+
+        response = self._make_http_request(
+            "GET",
+            urljoin(self.base_url, "vocabularies/{}/".format(id))
+        )
+
+        return response.json()
+
+    def _create_custom_vocabularies_options_payload(
+            self,
+            custom_vocabularies,
+            callback_url=None,
+            metadata=None):
+        payload = {}
+        if callback_url:
+            payload['callback_url'] = callback_url
+        if metadata:
+            payload['metadata'] = metadata
+        if custom_vocabularies:
+            payload['custom_vocabularies'] = custom_vocabularies
+        return payload
+
+    def _make_http_request(self, method, url, **kwargs):
```

Comment:
As I mentioned above. I think it would be a good idea to either make a base class or a class which both this and the `apiclient` can depend on which handles the actual creation of api requests (creating the auth header, disposing the http session). We do something similar in the node sdk here: https://github.com/revdotcom/revai-node-sdk/blob/develop/src/api-request-handler.ts

Benefits of this are: reduced duplicate code and easier testing